Perl/syntax_test_perl.pl

```diff
@@ -1214,470 +2103,2419 @@ =head1 B<--param>
+  $! @! %! *! $ERRNO $OS_ERROR
+# ^ punctuation.definition.variable.perl
+# ^^ variable.language.perl
+#    ^ punctuation.definition.variable.perl
+#    ^^ variable.language.perl
+#       ^ punctuation.definition.variable.perl
+#       ^^ variable.language.perl
+#          ^ punctuation.definition.variable.perl
+#          ^^ variable.language.perl
+#             ^ punctuation.definition.variable.perl
+#             ^^^^^^ variable.language.perl
+#                    ^ punctuation.definition.variable.perl
+#                    ^^^^^^^^^ variable.language.perl
+  $@ @@ %@ *@ $EVAL_ERROR
+# ^ punctuation.definition.variable.perl
+# ^^ variable.language.perl
+#    ^ punctuation.definition.variable.perl
+#    ^^ variable.language.perl
+#       ^ punctuation.definition.variable.perl
+#       ^^ variable.language.perl
+#          ^ punctuation.definition.variable.perl
+#          ^^ variable.language.perl
+#             ^ punctuation.definition.variable.perl
+#             ^^^^^^^^^^^ variable.language.perl
+  $? @? %? *? $CHILD_ERROR
+# ^ punctuation.definition.variable.perl
+# ^^ variable.language.perl
+#    ^ punctuation.definition.variable.perl
+#    ^^ variable.language.perl
+#       ^ punctuation.definition.variable.perl
+#       ^^ variable.language.perl
+#          ^ punctuation.definition.variable.perl
+#          ^^ variable.language.perl
+#             ^ punctuation.definition.variable.perl
+#             ^^^^^^^^^^^^ variable.language.perl
+
+# Deprecated and removed variables
+
+  $#  # format printed numbers
+# ^^ variable.language.deprecated.perl
+# ^ punctuation.definition.variable.perl
+  $*  # enable multiline matching
+# ^^ variable.language.deprecated.perl
+# ^ punctuation.definition.variable.perl
+  $[  # index of the first element in an array
+# ^^ variable.language.deprecated.perl
+# ^ punctuation.definition.variable.perl
+
+###[ UNQUALIFIED VARIABLES ]##################################################
+
+  $_foo = "bar";
+# ^ punctuation.definition.variable.perl
+# ^^^^^ variable.other.readwrite.perl
+#       ^ keyword.operator.assignment.perl
+#         ^ punctuation.definition.string.begin.perl
+#         ^^^^^ meta.string.perl string.quoted.double.perl
+#             ^ punctuation.definition.string.end.perl
+#              ^ punctuation.terminator.statement.perl
+  $_foo = "bar";
+# ^ punctuation.definition.variable.perl
+# ^^^^^ variable.other.readwrite.perl
+#       ^ keyword.operator.assignment.perl
+#         ^ punctuation.definition.string.begin.perl
+#         ^^^^^ meta.string.perl string.quoted.double.perl
+#             ^ punctuation.definition.string.end.perl
+#              ^ punctuation.terminator.statement.perl
+  $foo{bar}
+# ^^^^ variable.other.readwrite.perl
+# ^ punctuation.definition.variable.perl
+#     ^ punctuation.section.item-access.begin.perl
+#     ^^^^^ meta.item-access.perl
+#      ^^^ constant.other.key.perl
+#         ^ punctuation.section.item-access.end.perl
+  $foo{bar()}
```

Comment:
I think `bar()` should also have `meta.function-call.perl` so that the hover popup where `bar` is defined can find the places it is used.

Reply:
I think, I can re-add the `meta.function-call` to the function identifier, but it is not possible for the whole expression including the arguments list, because HEREDOCs can be passed as an argument and would cause the `meta.function-call` to never end.